Dockerfile.full_ros_build

```diff
@@ -49,18 +47,30 @@ ENV NAV2_DEPS_DIR $ROS2_WS/src/navigation2_dependencies
 RUN mkdir -p $NAV2_DEPS_DIR
 RUN vcs import src < $NAV2_DIR/tools/ros2_dependencies.repos
 
-# install package dependencies
-RUN apt-get update && \
-    rosdep install -q -y \
-      --from-paths \
-        $ROS2_WS/src \
-        src \
-      --ignore-src --skip-keys rti-connext-dds-5.3.1 \
+# install nav2 dependencies
+RUN apt-get update && rosdep install -y \
+    --from-paths src \
+    --ignore-src \
+    --skip-keys "console_bridge fastcdr fastrtps libopensplice67 libopensplice69 rti-connext-dds-5.3.1 urdfdom_headers" \
     && rm -rf /var/lib/apt/lists/*
 
-# build entire workspace
-ARG CMAKE_BUILD_TYPE=Debug
+# build source
+ARG MIXINS="build-testing-on release"
 RUN colcon build \
       --symlink-install \
-      --cmake-args \
-        -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE \
+      --mixin \
+        $MIXINS \
+      --packages-up-to \
+        nav2_system_tests \
```

Comment:
I'm a little worried that at some point, we may end up with a package in the repo that nav2_system_tests doesn't depend on. Then it wouldn't get built by this line. Maybe we can do something like:

```
NAV2_PACKAGES=(cd src/navigation2 && colcon list -n)
RUN colcon build \
...
  --packages-up-to $NAV2_PACKAGES \
```

Reply:
> NAV2_PACKAGES=`cd src/navigation2 && colcon list -n`

Hmm. Didn't know one could scope colcon like that without a `src` directory or PWD not being the workspace root.

Reply:
Ok, @crdelsey , I've take your advice here.